### PR TITLE
test: bump vitest

### PR DESCRIPTION
**Pull request by holic**
pulled out of https://github.com/latticexyz/mud/pull/995

### CODE REVIEW

File: pnpm-lock.yaml [hunk 1]
```diff
@@ -1,4 +1,4 @@
-lockfileVersion: '6.1'
+lockfileVersion: '6.0'
```
Comment (holic) [inline pnpm-lock.yaml:1]: @dk1a did you do anything in particular to force the lockfile version? when I try to `pnpm install` or `pnpm install -- force`, I still get the old version, even though I'm on pnpm 8.6.2

Comment (holic) [reply pnpm-lock.yaml:1]: even clearing out `node_modules` and reinstalling doesn't seem to help

Comment (dk1a) [reply pnpm-lock.yaml:1]: no, pnpm sometimes suggests upgrading itself, so I just did `pnpm add -g pnpm`,
I actually had a reverse problem where I couldn't really get '6.0'
you could try removing pnpm and installing it again I guess

@alvrs @roninjin10 curious about your experience with lockfileVersion

Comment (roninjin10) [reply pnpm-lock.yaml:1]: Npm uninstall and reinstall pnpm is another thing to try.   Also if you use nvm or anything where you have more than one node version might need to check specific node versions global store

Comment (holic) [reply pnpm-lock.yaml:1]: I tried `pnpm store prune`, removing `pnpm-lock.yaml` and reinstalling, uninstalling `pnpm` and reinstalling, and still no luck.

File: integration-sandbox/pnpm-lock.yaml [hunk 2]
```diff
@@ -1,4 +1,4 @@
-lockfileVersion: '6.1'
+lockfileVersion: '6.0'
```
Comment (holic) [inline integration-sandbox/pnpm-lock.yaml:1]:
```suggestion
lockfileVersion: '6.1'
```

File: pnpm-lock.yaml (hunk 1 above)
Comment (holic) [inline pnpm-lock.yaml:1]:
```suggestion
lockfileVersion: '6.1'
```

**Review by alvrs — APPROVED** [review]

File: pnpm-lock.yaml (hunk 1 above)
Comment (holic) [reply pnpm-lock.yaml]: turns out it was this: 
image

Comment (roninjin10) [reply pnpm-lock.yaml]: Ah glad it was something that actually makes sense.   I was about to quit using pnpm when you mentioned reinstalling pnpm doesn't fix it.  

As an aside it's interesting how often pnpm changes their schema version compared to yarn which never changes it

**Pull request merged**